src/App.tsx

```diff
@@ -1,8 +1,98 @@
-import React from 'react';
-import './App.css';
+import React, { useState, useMemo, useCallback } from 'react';
+import './App.scss';
+import debounce from 'lodash/debounce';
+import { getPosts, getUsers, getComments } from './api/api';
+import PostList from './PostList/PostList';
 
-const App = () => (
-  <h1>Dynamic list of posts</h1>
-);
+const getFilteredPosts = (posts: Post[], query: string) => {
+  const searchText = query.toLowerCase();
+
+  return posts.filter((post: Post) => (post.title + post.body)
+    .toLowerCase()
+    .includes(searchText));
+};
+
+const App = () => {
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [query, setQuery] = useState<string>('');
+  const [filterQuery, setfilterQuery] = useState<string>('');
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isVisible, setIsVisible] = useState<boolean>(false);
+
+  const loadPosts = async () => {
+    setIsLoading(true);
+    setIsVisible(true);
+    const postFromServer = await getPosts();
+    const usersFromServer = await getUsers();
+    const commentsFromServer = await getComments();
+
+    const preparedPosts = postFromServer.map(post => {
+      const user = usersFromServer
+        .find((currentUser: User) => currentUser.id === post.userId);
+      const userComments = commentsFromServer
+        .filter((comment: Comment) => (comment.postId === post.id));
+
+      return {
+        ...post,
+        user,
+        userComments,
+      };
+    });
+
+    setPosts(preparedPosts);
+
+
+    setIsLoading(false);
+  };
+
+  const setfilterQueryWhitDebounce = useCallback(debounce(setfilterQuery, 1000), []);
+
+  const handleChange = (event: { target: { value: React.SetStateAction<string> } }) => {
+    setfilterQueryWhitDebounce(event.target.value);
+    setQuery(event.target.value);
```

Comment:
А ты value не достал?
```suggestion
    setfilterQueryWhitDebounce(value);
    setQuery(value);
```